File: typescript/1022.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function sumRootToLeaf(root: TreeNode | null): number {
    function getNodeSum(node: TreeNode | null, s: string = ""): number {
        if (node == null) return 0;

        let newS = s;
        newS += node.val;

        if(node.left == null && node.right == null) return parseInt(newS, 2);
        
        return getNodeSum(node.left, newS) + getNodeSum(node.right, newS)
    }

    return getNodeSum(root)
};

File: kotlin/1022.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {

    fun getNodeSum(node: TreeNode?, s: String = ""): Int {
        if (node == null) return 0

        val newS = s + node.`val`

        if (node.left == null && node.right == null) {
            return newS.toInt(2)
        }

        return getNodeSum(node.left, newS) + getNodeSum(node.right, newS)
    }

    fun sumRootToLeaf(root: TreeNode?): Int {
        return getNodeSum(root)
    }
}